### PR TITLE
Add ag grid v32 support

### PR DESCRIPTION
**Pull request by origami-z**
On top of #4257

Upgrade to v32 to support 

- New active filter icon style
- New default menu style (not tabbed menu), see [doc](https://www.ag-grid.com/react-data-grid/column-menu/#legacy-tabbed-column-menu)

Todos

- [x] filter icon active state (both in floating filter and column header cell), when filter is active, need to show some visual indicator (e.g. [ag grid examples](https://www.ag-grid.com/example/) shows a light blue background and dot badge). In this PR, it's implemented match closely to Salt toggle button active state.  #4275
    - [x] MTK feedback on different filter icon used (clear vs filled). ~~Preview available in #4335~~
- [x] filter panel apply (blue) / reset (gray) button, changed from `ref` attribute to `data-ref`. 
- [x] header sort icon should be next to filter / menu icon
    - [x] double check v31 regression
- [x] checkbox column (e.g. context menu story) shouldn't have filter
- [x] context menu size larger due to `ag-size` change
- [x] ~~mention `suppressMenuHide` [Customising the Column Menu](https://www.ag-grid.com/javascript-data-grid/column-menu/#customising-the-column-menu)~~ added to one site example
---

Snapshot test on StackBlitz

- v31 backwards compatibility: https://stackblitz.com/edit/salt-ag-grid-v31-theme-v2-oxpnan?file=package.json,App.jsx
- v32 new upgrade: https://stackblitz.com/edit/salt-ag-grid-v32-theme-v2-5m5lcf?file=package.json,App.jsx

### CODE REVIEW

File: packages/ag-grid-theme/css/parts/ag-root-var.css [hunk 1]
```diff
@@ -16,7 +16,7 @@ div[class*="ag-theme-salt"] {
   --ag-font-family: var(--salt-text-fontFamily);
   --ag-font-size: var(--salt-text-fontSize);
   --ag-foreground-color: var(--salt-content-primary-foreground);
-  --ag-grid-size: var(--salt-spacing-50);
+  --ag-grid-size: var(--salt-spacing-100);
```
Comment (origami-z) [inline packages/ag-grid-theme/css/parts/ag-root-var.css:19]: TODO: this impacts menu sizes, which is not wanted. This was changed for filter and menu icon spacing on header cells. need a different solution

Comment (origami-z) [reply packages/ag-grid-theme/css/parts/ag-root-var.css:19]:
```
.ag-ltr .ag-header-cell:not(.ag-right-aligned-header) .ag-header-label-icon, .ag-ltr .ag-header-cell:not(.ag-right-aligned-header) .ag-header-menu-icon {
    margin-left: var(--ag-grid-size);
}
```

**Review by bhoppers2008 — APPROVED** [review]

File: yarn.lock [hunk 2]
```diff
@@ -17345,13 +17364,20 @@ __metadata:
...
 
-"tabbable@npm:^6.0.0, tabbable@npm:^6.0.1":
+"tabbable@npm:^6.0.0":
   version: 6.2.0
   resolution: "tabbable@npm:6.2.0"
   checksum: 10/980fa73476026e99dcacfc0d6e000d41d42c8e670faf4682496d30c625495e412c4369694f2a15cf1e5252d22de3c396f2b62edbe8d60b5dadc40d09e3f2dde3
   languageName: node
   linkType: hard
 
+"tabbable@npm:^6.0.1":
```
Comment (joshwooding) [inline yarn.lock:17374]: `yarn dedupe` should clean up the dependencies.

**Review by Fercas123 — APPROVED** [review]

**Review by bhoppers2008 — APPROVED** [review]

**Pull request merged**